doc/verification.rst

```diff
@@ -5,27 +5,63 @@ Overview
 --------
 
 This is a SV/UVM testbench for the ibex core verification.
-This testbench loads the instruction binary generated from open source random instruction generator `riscv-dv <https://github.com/google/riscv-dv>`_, runs the RTL simulation, and compares the instruction trace against ISS simulation.
+At a high level, this testbench uses the open source `RISCV-DV random instruction generator
+<https://github.com/google/riscv-dv>`_ to generate compiled instruction binaries, loads them into a
+simple memory mode, stimulates the Ibex core to run this program in memory, and then compares the
+core trace log against a golden model ISS trace log to check for correctness of execution.
 
-Testbench component
-~~~~~~~~~~~~~~~~~~~
+Testbench Architecture
+----------------------
 
--  Random instruction generator : `RISCV-DV <https://github.com/google/riscv-dv>`_
--  `Memory interface agent for instruction fetch and load/store
-   operations <https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/ibex_mem_intf_agent>`_
--  `Interrupt interface agent <https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/irq_agent>`_
--  `Memory model <https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/mem_model>`_
--  `Test and sequence library <https://github.com/lowRISC/ibex/tree/master/dv/uvm/tests>`_
+As previously mentioned, this testbench has been constructed based on its usage of the RISCV-DV
+random instruction generator developed by Google. A block diagram of the testbench is below.
+
+.. figure:: images/tb.svg
+
+Memory Interface Agents
+~~~~~~~~~~~~~~~~~~~~~~
+
+Code can be seen `here
+<https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/ibex_mem_intf_agent>`_.
+Two of these agents are instantiated within the testbench, one for the instruction fetch interface,
+and the second for the LSU interface. These agents run slave sequences that wait for memory requests from
+the core, and then grant the requests for instructions or for data.
+
+Interrupt Interface Agent
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Code can be seen `here <https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/irq_agent>`_.
```

Comment:
Please avoid linking the word "here", as it doesn't work well if someone cannot click on the link (e.g. because they're reading the PDF). Instead, write something like this:

```rst
The code can be found in the `dv/uvm/common/irq_agent <https://github.com/lowRISC/ibex/tree/master/dv/uvm/common/irq_agent>`_ directory.
```

Use the same approach for the other sections below.

Reply:
Done, all the links have been updated to be more clear.